src/input_output/FGPropertyManager.h

```diff
@@ -62,6 +62,37 @@ template<typename T> T getValue(const SGPropertyNode* node)
   return static_cast<T>(node->getIntValue());
 }
 
+template <class C, class T>
+class SGRawValueMethodsEnum : public SGRawValue<int>
+{
+public:
+  typedef T(C::* getter_t)() const;
+  typedef void (C::* setter_t)(T);
+  SGRawValueMethodsEnum(C& obj,
+    getter_t getter = 0, setter_t setter = 0)
+    : _obj(obj), _getter(getter), _setter(setter) {
+  }
+  virtual ~SGRawValueMethodsEnum() {}
+  virtual int getValue() const {
+    if (_getter) { return (int)(_obj.*_getter)(); }
+    else { return SGRawValue<int>::DefaultValue(); }
+  }
+  virtual bool setValue(int value) {
+      return this->setValue((T)value);
+  }
+  bool setValue(T value) {
+    if (_setter) { (_obj.*_setter)(value); return true; }
+    else return false;
+  }
+  virtual SGRaw* clone() const {
+    return new SGRawValueMethodsEnum(_obj, _getter, _setter);
+  }
+private:
+  C& _obj;
+  getter_t _getter;
+  setter_t _setter;
+};
+
 template <class C, class T, class U>
 class SGRawValueMethodsIndexedEnum : public SGRawValue<T>
```

Comment:
Would not the class `SGRawValueMethodsIndexedEnum` need the same mechanism ?
```suggestion
class SGRawValueMethodsIndexedEnum : public SGRawValue<int>
```

Reply:
Only if the data being returned is an Enum type (type `T`). When using the Enum only as an index type (type `U`), the inheritance is still appropriate to come from `SGRawValue<T>`